RadarSDK/RadarSettings.m

```diff
@@ -126,36 +127,78 @@ + (void)setTracking:(BOOL)tracking {
 
 + (RadarTrackingOptions *)trackingOptions {
     NSDictionary *optionsDict = [[NSUserDefaults standardUserDefaults] dictionaryForKey:kTrackingOptions];
-    return [RadarTrackingOptions trackingOptionsFromDictionary:optionsDict];
+
+    if (optionsDict != nil) {
+        return [RadarTrackingOptions trackingOptionsFromDictionary:optionsDict];
+    } else {
+        return nil;
+    }
 }
 
 + (void)setTrackingOptions:(RadarTrackingOptions *)options {
     NSDictionary *optionsDict = [options dictionaryValue];
     [[NSUserDefaults standardUserDefaults] setObject:optionsDict forKey:kTrackingOptions];
 }
 
-+ (void)removeRemoteTrackingOptions {
-    [[NSUserDefaults standardUserDefaults] removeObjectForKey:kRemoteTrackingOptions];
++ (void)removeTrackingOptions {
+    [[NSUserDefaults standardUserDefaults] removeObjectForKey:kTrackingOptions];
+}
+
++ (RadarTrackingOptions *)previousTrackingOptions {
+    NSDictionary *optionsDict = [[NSUserDefaults standardUserDefaults] dictionaryForKey:kPreviousTrackingOptions];
+
+    if (optionsDict != nil) {
+        return [RadarTrackingOptions trackingOptionsFromDictionary:optionsDict];
+    } else {
+        return nil;
+    }
+}
+
++ (void)setPreviousTrackingOptions:(RadarTrackingOptions *)options {
+    NSDictionary *optionsDict = [options dictionaryValue];
+    [[NSUserDefaults standardUserDefaults] setObject:optionsDict forKey:kPreviousTrackingOptions];
+}
+
++ (void)removePreviousTrackingOptions {
+    [[NSUserDefaults standardUserDefaults] removeObjectForKey:kPreviousTrackingOptions];
 }
 
 + (RadarTrackingOptions *_Nullable)remoteTrackingOptions {
     NSDictionary *optionsDict = [[NSUserDefaults standardUserDefaults] dictionaryForKey:kRemoteTrackingOptions];
-    return optionsDict ? [RadarTrackingOptions trackingOptionsFromDictionary:optionsDict] : nil;
+
+    if (optionsDict != nil) {
+        return optionsDict ? [RadarTrackingOptions trackingOptionsFromDictionary:optionsDict] : nil;
```

Comment:
Technically unnecessary since I added the `nil` check in `+[RadarTrackingOptions trackingOptionsFromDictionary:]`, but I think that this is a little clearer.